lib/puppet/file_serving/http_metadata.rb

```diff
@@ -0,0 +1,21 @@
+require 'puppet/file_serving/metadata'
+
+# Simplified metadata representation, suitable for the information
+# that is available from HTTP headers.
+class Puppet::FileServing::HttpMetadata < Puppet::FileServing::Metadata
+
+  def initialize(http_response, path = '/dev/null')
```

Comment:
Thumbing through the source code for `FileServing::Metadata` and it's parent class, it wasn't immediately obvious to me if there was any functionality in those that you'd want/need to inherit here?  I suppose the `to_data_hash` method?  If that's the only one, it might be worth considering just implementing that in this class directly, rather than subclassing.  It seems like most of the logic in the parent classes is about stat'ing the filesystem, etc., which seems like stuff that we'd want to avoid for this implementation.

Not sure whether this is a good idea or not, just thinking out loud.  I might try attaching a debugger and poking around to see what stuff from the parent classes gets called when we're going through `HttpMetadata`.  Will post a follow-up comment if that yields anything interesting.